### ISSUE TITLE
BW Error when running 110 Node run

### ISSUE THREAD

**Issue by jacobrh91** [jacobrh91/Swift-T-Variant-Calling#30]
I have successfully tested a job with 2 nodes on BW, but when I scale up to running 200 samples on 110 Nodes (using 20 ADLB_SERVERS, two on each of the extra 10 Nodes)

I get an error that complains that a samtools view command encountered a truncated file. Since this occurs directly after some reads have finished aligning, this is a key sign that one of the checkBam functions is evaluating an output bam before it has actually finished being written to.

Poking through the logs and output files, I have discovered that out of the 200 samples being analyzed, there is only one sample that does not have an <aligned>.bam file (the SRR1438242/align directory was empty)

In addition, I have poked through the alignment logs, and while most of the bwa logs end with 
```
[main] Version: 0.7.12-r1039
[main] CMD: /projects/sciteam/baib/builds/bwa/bwa-0.7.12/bwa mem -M -k 32 -I 300,30 -t 16 -R @RG\tID:SRR1531512\tLB:\tPL:ILLUMINA\tPU:SRR1531512\tSM:SRR1531512\tCN: /projects/sciteam/baib/GATKbundle/July1_2017/LSM_July1_2017/human_g1k_v37_decoy.SimpleChromosomeNaming.fasta /scratch/sciteam/jacobrh/purge_exempt/fastq_batches/batch3/SRR1531512_1.fastq.gz /scratch/sciteam/jacobrh/purge_exempt/fastq_batches/batch3/SRR1531512_2.fastq.gz
```

which signals that the bwa program finished correctly, there was one file that did not have this completed signature. (logs/SRR1438242_Alignment.log)

So, my conclusion is that there is one alignment that is not finished, but Swift/T tries to go on, even though there is an implicit (and now explicit, i.e. => command at the end of the alignment line, command added to the source code)

My only guess at this point as to what is going wrong is that I am using too many ADLB_SERVERS for the task at hand, and they are having trouble communicating/coordinating with each other. If I roll this back to only having 1 ADLB_SERVER again, maybe this will fix the problem...

(Update)

It turns out that the actual command that killed the job was a samtools call on a different sample! Here is the actual error message in the Swift/T log:

```
14515 CAUGHT ERROR:^M
14516 [E::bgzf_read] Read block operation failed with error -1 after 0 of 4 bytes^M
14517 [main_samview] truncated file.^M
14518     while executing^M
14519 "exec $samtoolsexe view -c $inputFile "^M
14520     (procedure "alignment::samtools_view" line 3)^M
14521     invoked from within^M
14522 "alignment::samtools_view ${v:samtoolsexe} ${v:of:bamFile} "^M
14523     (procedure "samtools_view2-argwait" line 6)^M
14524     invoked from within^M
14525 "samtools_view2-argwait 5487 /scratch/sciteam/jacobrh/purge_exempt/Swift_testing/110Nodes/110Node_output/SRR1284952/align/SRR1284952.noDedups.bam 3701"^M
14526 Turbine worker task error in: samtools_view2-argwait 5487 /scratch/sciteam/jacobrh/purge_exempt/Swift_testing/110Nodes/110Node_output/SRR1284952/align/SRR1284952.noDedups.bam 3701^M
14527     invoked from within^M
14528 "c::worker_loop $WORK_TYPE($mode) $keyword_args"^M
14529     (procedure "standard_worker" line 27)^M
14530     invoked from within^M
14531 "standard_worker $rules $startup_cmd "^M
14532     (procedure "enter_mode_unchecked" line 5)^M
14533     invoked from within^M
14534 "enter_mode_unchecked $rules $startup_cmd"^M
14535     (procedure "enter_mode" line 10)^M
14536     invoked from within^M
14537 "enter_mode $rules $startup_cmd "^M
14538 ADLB: ADLB_Abort(1) calling MPI_Abort(1)
14539 Application 63427736 exit codes: 134
14540 Application 63427736 exit signals: Killed
14541 Application 63427736 resources: utime ~995s, stime ~361s, Rss ~17640, inblocks ~19944652, outblocks ~1329
```

**Comment by j-woz** [comment]
Is it possible that the end of the file has not yet been flushed to the filesystem when the next task starts?  Do you have timestamps that would show the task ordering?  
Can you point me to the part of the code that has the implicit or explicit dependency?

**Comment by azzaea** [comment]
Not sure if this would be helpful, but thought to think out loud.

@jacobrh91 : Probably it helps to see timings via the logging app? Since this time the workflow run didn't reach completion, the `Timing.log` would only have partial entries. However, the small files in the `SWIFT_TMP` directory would contain the intermediate timing logs.
You can then just cat them all to a single file, and then upload this file to the logging app as per: https://github.co/ncsa/Swift-T-Variant-Calling#logging-functionality

It would at least give you an idea of what ran when, and ultimately tracing the logs should be easier.

**Comment by lsmainzer** [comment]
Azza: good idea. That was indeed the original goal of building the timing logs.

**Comment by jacobrh91** [comment]
@zo0z: When I try to use the R thing, an error pops up that says "ERROR: there is no package called 'shinydashboard'" but when looking at the 'cat'ed logs I found a few things.

However, I finally got a 110 Node run to work when I switched from using Picard to Novosort for marking duplicates. But I also changed from using 20 ADLB_SERVERS to uusing 1, and changed the ppn from 2 to 1. 

I'm rerunning a picard run with the last two changes still present to see if it was actually one of them that solved the problem.

I'll close this ticket now, but I may add the details of the picard test later.
